babar-agent/src/main/java/com/criteo/babar/agent/profiler/utils/ProcFSUtils.java

```diff
@@ -174,7 +174,55 @@ protected static ProcPidIO parsePidIO(String output) {
         return new ProcPidIO(rchar, wchar, readBytes, writeBytes);
     }
 
-    public static class ProcPidStat {
+    public static ProcNetIO netio() throws IOException {
+        String outputProcNetDev = OSUtils.exec(new String[]{"cat", "/proc/net/dev"});
+        String outputSysClassNet = OSUtils.exec(new String[]{"ls", "-l", "/sys/class/net"});
+        return parseNetIO(outputProcNetDev, outputSysClassNet);
+    }
+
+    protected static ProcNetIO parseNetIO(String outputProcNetDev, String outputSysClassNet) {
+    	List<String> realDevices = parseRealDevices(outputSysClassNet);
+    	String[] lines = outputProcNetDev.split("\n");
+        if (!lines[0].startsWith("Inter-|")) {
+            throw new RuntimeException("Unexpected first line in output of /proc/dev/net:\n" + outputProcNetDev);
+        }
+
+        if (lines.length < 2) {
+            throw new RuntimeException("Not enough lines in output of /proc/dev/net");
+        }
+
+        long rxBytes = 0;
+        long txBytes = 0;
+
+        for(int i=2; i < lines.length; i++) {
+            String[] cols = lines[i].trim().split(" +"); //Trim leading spaces
+
+            if(cols.length == 17 && realDevices.contains(cols[0])) {
+                rxBytes += Long.parseLong(cols[1]);
+                txBytes += Long.parseLong(cols[9]);
+            }
+        }
+
+        return new ProcNetIO(rxBytes, txBytes);
+	}
+
+	protected static List<String> parseRealDevices(String outputSysClassNet) {
+		List<String> realDevices = new LinkedList<>();
```

Comment:
I see that we then check for a device using `contains()`, for this reason I would avoid using a `LinkedList` and instead use an `ArrayList` (which will be better for cache locality and will avoid having to allocate an iterator to go through the list).
Even better, as we will usually have more than a 2-3 devices, we could use a `HashSet` to allow searching in `O(1)`